Jint/Extensions/ReflectionExtensions.cs

```diff
@@ -54,5 +58,108 @@ private static bool IsExtensionMethod(this MethodBase methodInfo)
         {
             return methodInfo.IsDefined(typeof(ExtensionAttribute), true);
         }
+
+        public static bool IsNullable(this Type type)
+        {
+            return type is { IsGenericType: true } && type.GetGenericTypeDefinition() == nullableType;
+        }
+
+        public static bool IsNumeric(this Type type)
+        {
+            if (type == null || type.IsEnum)
+            {
+                return false;
+            }
+
+            switch (Type.GetTypeCode(type))
+            {
+                case TypeCode.Byte:
+                case TypeCode.Decimal:
+                case TypeCode.Double:
+                case TypeCode.Int16:
+                case TypeCode.Int32:
+                case TypeCode.Int64:
+                case TypeCode.SByte:
+                case TypeCode.Single:
+                case TypeCode.UInt16:
+                case TypeCode.UInt32:
+                case TypeCode.UInt64:
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
+        public static bool IsClrNumericCoercible(this Type type)
+        {
+            if (type == null || type.IsEnum)
+            {
+                return false;
+            }
+
+            switch (Type.GetTypeCode(type))
+            {
+                case TypeCode.Decimal:
+                case TypeCode.Double:
+                case TypeCode.Int32:
```

Comment:
Not the smaller ones?

Reply:
My thinking was less surprises, like overflows etc. `int` should be quite common and we could open up later on if someone has the use case.